### PR TITLE
Feature/Heap

### PR DESCRIPTION
**Pull request by ProCode2**
Feature: Added Heap Implementation with slices
Supperted ops: `add`, `peek`, `poll` and other base operations

### CODE REVIEW

File: heap/heap.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+	"github.com/wwwangxc/container/heap/minheap"
+	"golang.org/x/exp/constraints"
+)
+
+// Return a new minheap
+func NewMinHeap[T constraints.Ordered](elements ...T) container.Heap[T] {
+	return minheap.New(elements...)
+}
+
+// Return a nee maxheap
```
Comment (wwwangxc) [inline heap/heap.go]: nee -> new

File: heap/maxheap/container.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+// Remove all elements from the heap
+func (mh *MaxHeap[T]) Clear() {
+    mh.size = 0;
+    mh.elements = []T{}
+}
+
+// Return all the elements of the heap
+func (mh *MaxHeap[T]) Values() []T {
+    if mh.size == 0 {
+		return []T{}
```
Comment (wwwangxc) [inline heap/maxheap/container.go:27]: please keep the correct indentation

File: heap/heap.go (hunk 1 above)
Comment (ProCode2) [reply heap/heap.go]: oops, updated!

File: heap/maxheap/container.go (hunk 2 above)
Comment (ProCode2) [reply heap/maxheap/container.go:27]: Somehow my editor forgot about that line, updated.

**Comment by wwwangxc** [comment]
LGTM

**Comment by wwwangxc** [comment]
Thanks for your contribution.

**Pull request merged**